app/src/main/java/io/github/dkter/aaaaa/AaaaaKeyboardView.kt

```diff
@@ -96,17 +104,16 @@ class AaaaaKeyboardView(
         if (this.getBooleanPref(R.string.hapticFeedbackKey))
             v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY)
 
-        if (id == R.id.btnA) {
-            this.keyboardListener.onA()
-        }
-        else if (id == R.id.btnBackspace) {
-            this.keyboardListener.onBackspace()
-        }
-        else if (id == R.id.btnSpace) {
-            this.keyboardListener.onSpace()
-        }
-        else if (id == R.id.btnReturn) {
-            this.keyboardListener.onReturn()
+        when (id) {
+            R.id.btnA -> keyboardListener.onA()
+            R.id.btnBackspace -> keyboardListener.onBackspace()
+            R.id.btnSpace -> keyboardListener.onSpace()
+            R.id.btnReturn -> keyboardListener.onReturn()
+            R.id.btnUppercase -> {
+                isUppercase = !isUppercase
+                btnA.text = if (isUppercase) "A" else "a"
+                if (isUppercase) keyboardListener.onUppercase() else keyboardListener.onLowercase()
```

Comment:
Two things:
1. The setting of the caps lock button state (changing A button label and setting tint) only happens when the caps lock button is pressed. I was able to get the state to be wrong by rotating my phone as well as dismissing the keyboard and re-showing it (the button was grey, the keyboard was lowercase but it typed uppercase As anyway). I'd move this stuff to its own method and call it when the keyboard pops up or something.
2. Now that I'm on this topic, it seems a bit weird to have the caps lock button state persist across the showing and hiding of the keyboard. When the keyboard gets dismissed, I would reset `isUppercase` to false.

Reply:
Is ther a callback I can use from within `AaaaaKeyboardView` for when the keyboard gets dismissed? I haven't found anything.

Reply:
In `AaaaaKeyboardView` it looks like everything is already reset when the keyboard is dismissed (since every time the keyboard is started a new `AaaaaKeyboardView` is created). The `isUppercase` that needs to be reset is the one in `AaaaaInputMethodService` and that can be done in `AaaaaInputMethodService.onFinishInputView`.

`AaaaaKeyboardView.isUppercase` should also probably be set to `AaaaaInputMethodService.isUppercase` on `init`. Maybe move `isUppercase` to `AaaaaKeyboardListener` to make that property accessible.

Reply:
Ah, you're right. I set `isUppercase` to reset on keyboard close in 6ea9ff1. 

> AaaaaKeyboardView.isUppercase should also probably be set to AaaaaInputMethodService.isUppercase on init. Maybe move isUppercase to AaaaaKeyboardListener to make that property accessible.

I don't see the need since it will always be false. Am I missing something?

Reply:
There isn't an urgent need for it now but I feel like it could help prevent future bugs.

Reply:
You're right. Added in 01f82ab.